server/src/main/java/org/opensearch/search/aggregations/bucket/histogram/DateHistogramAggregator.java

```diff
@@ -201,6 +226,79 @@ public void collect(int doc, long owningBucketOrd) throws IOException {
         };
     }
 
+    private String fetchStarTreeCalendarUnit() {
+        CompositeDataCubeFieldType compositeMappedFieldType = (CompositeDataCubeFieldType) context.mapperService()
+            .getCompositeFieldTypes()
+            .iterator()
+            .next();
+        DateDimension starTreeDateDimension = (DateDimension) compositeMappedFieldType.getDimensions().get(0);
```

Comment:
I don't entirely understand what this method is doing, especially with the side-effect of changing `starTreeDateRoundingRequired`.

Since it's called from the constructor, I think it would be clearer to add a small immutable static inner class that holds the `dimensionName` and `starTreeDateRoundingRequired`. Then it could be a `final` variable. (Once we upgrade to Java 21, it could be a `record`.)